core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -230,7 +230,7 @@ protected void parse() throws ProtocolException {
         time = readUint32();
         difficultyTarget = readUint32();
         nonce = readUint32();
-        hash = Sha256Hash.wrapReversed(Sha256Hash.hashTwice(payload, offset, cursor));
+        hash = Sha256Hash.wrapReversed(Sha256Hash.hashTwice(payload, offset, cursor - offset));
```

Comment:
This doesn't seem right. A couple of lines above, there is "cursor = offset", so that cursor-offset would always be 0.